### PR TITLE
Add undo feature

### PR DESCRIPTION
**Pull request by tanchinkiat99**
Added undo feature to address Issue #106 

Undo feature undoes the previous executed Add, Delete or Edit command. Each of these commands will save the current version of the address book inside the AddressBook object before it is executed, and Undo Command will retrieve this saved version. 

Unit testing has been executed to ensure its success in undoing the above 3 commands.

### CODE REVIEW

**Review by cwq2326 — APPROVED** [review]
LGTM 💯

File: src/test/java/seedu/address/logic/commands/UndoCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.TypicalPersons;
+
+
```
Comment (jiale-c) [inline src/test/java/seedu/address/logic/commands/UndoCommandTest.java:20]: LGTM! Maybe you can add a descriptor for this test class to talk about the type of testing done here.

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Erases the previous change to the address book and restores the the older state.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_UNDO_SUCCESS = "Undid last change.";
+
```
Comment (jiale-c) [inline src/main/java/seedu/address/logic/commands/UndoCommand.java:20]: Can add another message if the undo failed? For example just launching the app and using undo immediately.:grin:

**Review by jiale-c — APPROVED** [review]
LGTM! I have highlighted some comments too!

**Pull request merged**